expand pronoun explanation and add resources (#1411)
carpentries/instructor-training: _episodes/01-welcome.md

> 
> Note that pronouns are personal and some participants might prefer not to share them.
> Do not force people to share their pronouns.
> One reason to avoid pressuring people to share them is to allow people to share their 
> gender identity only when they feel ready to. 

> > It is however necessary for all participants to use the provided pronouns and names when available. 

Reviewer: ```suggestion
> It is, however, necessary for all participants to use pronouns and names as listed when participants provide them. 
```
